### ISSUE TITLE
WAT not showing up in IE 11

### ISSUE THREAD

**Issue by AnandiNa** [ThePacielloGroup/WebAccessibilityToolbar#96]
WAT says it's successfully instlled, but it does not show up in IE 11ool bars..I have uninstalled and reinstalled several times. Can you please help?

Thanks.

**Comment by menovak** [comment]
Hi AnandiNa,

Can you please provide a bit more detail.

Are you installing on Windows 7 or 8, and as 32 bit or 64 bit?

Please use the latest download from Github (
https://github.com/ThePacielloGroup/WebAccessibilityToolbar/releases/tag/2015-05-20)
noting that you must install the proper 32 or 64/32 bit version.

Also, many times once WAT is installed, based on security settings you
sometimes need to 'enable' the toolbar to 'see' it.   Look under IE's
menus, View -> Toolbars.

Finally the last item to check, if none of the above fix your problem, look
under Tools -> Internet Options -> Programs -> Manage add-ons, and be sure
the WAT toolbar is enabled under the 'status' column. (right click can be
used to change status)

Please let me know if this helps.

Regards,

Mark
TPG, Engineer

**Comment by AnandiNa** [comment]
Hi:
Thanks for the prompt response.
I am using Windows 7 64 bit OS.
I tried again with the latest download link that you provided and it is still the same..

I couldn’t see “enable” under IE’s View ->tool bars
[cid:image001.png@01D0C953.9DA81010]

Finally, status column on manage add-ons is “New” and both enable and disable are disabled.

[cid:image003.png@01D0C953.F63E9730]



Also, this is not only for me.. Many people In my organization have been facing this issue.  We had figured out a workaround ,(uninstall IE 11, install IE 10, and then install WAT and then upgrade to IE 11). That used to work for some time. Now, even that is not working, because  installing IE 10 and then upgrading does  not work like before.

It will be great if we can find a permanent solution for this problem.

Thanks.
Anandi

**Comment by menovak** [comment]
hi Anandi,

I think you tried to attach a picture, but it did not transfer...

I do not have 64 bit Windows 7, so I've CC'd  colleague, perhaps he can
check.

1 - under View->Toolbars, there is no enable, just a check box so that WAT
shows up.  See attached.  Do you even see WAT under this menu?

2 - under manage add ons, you can see in attached, what you should have
when WAT is running in 64 bit on Windows 8.1.  Does WAT show up here?

Sorry to hear others within your organization are also having issues.  The
most recent WAT was tested on 32/64 bit systems, from different people, and
no one suggested an issue during install when the correct mode was chosen.
Perhaps there is an issue, which is going to take some trouble-shooting.

As always, there could also be a conflicting other 'toolbar' or application
add-on loaded, so that is another option to try, if you have not already
done so, to unload all other add-ons... (or disabled them).

Mark

**Comment by patrickhlauke** [comment]
Wondering if this may be related to some Group Policy that was set for all users in the organisation? See for instance http://community.spiceworks.com/how_to/1428-use-gpo-to-disable-those-annoying-toolbars-in-internet-explorer

**Comment by menovak** [comment]
Hi Anandi

On Wed, Jul 29, 2015 at 10:17 AM, AnandiNa <notifications@github.com> wrote:

>
>
> From: menovak [mailto:notifications@github.com]
> Sent: Tuesday, July 28, 2015 5:15 PM
> To: ThePacielloGroup/WebAccessibilityToolbar <
> WebAccessibilityToolbar@noreply.github.com>
> Cc: Narasimhan, Anandi (CDC/ONDIEH/NCCDPHP) (CTR) <idl1@cdc.gov>
> Subject: Re: [WebAccessibilityToolbar] WAT not showing up in IE 11 (#96)
>
> hi Anandi,
>
> I think you tried to attach a picture, but it did not transfer...
>
> I do not have 64 bit Windows 7, so I've CC'd colleague, perhaps he can
> check.
>
> 1 - under View->Toolbars, there is no enable, just a check box so that WAT
> shows up. See attached. Do you even see WAT under this menu?
> Looks like the attachment has been stripped off..I don’t see WAT under
> View->Toolbars
>
>

I've attached as jpeg this time, hope that works



>
> 2 - under manage add ons, you can see in attached, what you should have
> when WAT is running in 64 bit on Windows 8.1. Does WAT show up here?
> Again, there is not attachment. But WAT is available under “not Verified”
> section. Also, enable and disable both buttons are disabled.
>



I've attached as jpeg this time, hope that works

If you see WAT in the 'manage add-ons', I think that means it is loaded but
something is preventing it from being 'enabled'.  Have you tried unloading
or disabling all other toolbars one by one ?

Can you send me a screen shot of what you have in 'manage add-ons'?

Note, WAT needs to install a DLL, perhaps that is not allowed?

A colleague also sent me this link, can you check to be sure you do not
have some kind of policy where you work which prevents  using toolbars, or
in our case 'not verified' software ?


"""Wondering if this may be related to some Group Policy that was set for
all users in the organisation? See for instance
http://community.spiceworks.com/how_to/1428-use-gpo-to-disable-those-annoying-toolbars-in-internet-explorer
  """





>
> Sorry to hear others within your organization are also having issues. The
> most recent WAT was tested on 32/64 bit systems, from different people,
> and
> no one suggested an issue during install when the correct mode was chosen.
> Perhaps there is an issue, which is going to take some trouble-shooting.
>
> As always, there could also be a conflicting other 'toolbar' or
> application
> add-on loaded, so that is another option to try, if you have not already
> done so, to unload all other add-ons... (or disabled them).
>
> Mark
>
>
> On Tue, Jul 28, 2015 at 3:39 PM, AnandiNa <notifications@github.com

**Comment by patrickhlauke** [comment]
folks, mind using github to answer? these email responses are not coming through properly here...

**Comment by AnandiNa** [comment]
I have checked with my network manager and he  would like to know what group policies are being opened to install this toolbar. Can you please let us know? Also, it has installed for some people in our team before.. So, I am not sure if group policies is the answer.

**Comment by menovak** [comment]
Hi Anandi,

I am not involved with the installer development, therefore I will have to ask and get back to you.  I'm doubtful it is a 'group policy' issue either, but good to check.  

Have you removed any other toolbars to see if there was a possible toolbar conflict?

Mark

**Comment by menovak** [comment]
Hey Matt or Jun,

Any ideas as to what, if any, "group policies" might be opened by the WAT
installer?

Thanks

Mark

**Comment by matatk** [comment]
It uses Inno Setup and I believe the default permissions required will be assumed to be [admin](http://www.jrsoftware.org/ishelp/index.php?topic=setup_privilegesrequired), though I’m not certain (as the setup file does not explicitly specify this).

From the [setup script itself](https://github.com/ThePacielloGroup/WebAccessibilityToolbar/blob/master/Web-Accessibility-Toolbar.iss#L64-L82) it does need access to Program Files, so it probably does need some sort of elevated privileges, though I'm not au fait enough to know what the minimal set of permissions required would be.

**Comment by menovak** [comment]
hi Anandi,

Talked with one of our other developers and he said that "WAT uses Inno Setup and the default permissions required will be assumed to be 'admin' ".  Waiting to hear from one other person...

In the meantime, just wondering if you've tried to install WAT by right clicking on the icon and select
'Run as administrator' ?

Mark

**Comment by AnandiNa** [comment]
We have no other toolbars. Also, I tried installing with  "run as admin". Now, I can see it when I right-click the toolbar area (that is the only difference, but it's not getting checked and so it is not showing up in the toolbars.

**Comment by matatk** [comment]
When it's installed for the first time, IE will put a pop-up (at the very bottom of the screen) to ask the user to enable it (or not) -- I wonder if this has been missed, or answered with "don't enable" at some point in the past?  Does it now show up in the Add-on Manager (Tools -> Internet Options -> Programs -> Manage add-ons), as well as the available toolbars list?

**Comment by bpace5** [comment]
Anandina, 

Have your network manager take a look at this in order to enable WAT Toolbar for your organization. Picture attached.

Brandon
![2015-11-16_13-34-37](https://cloud.githubusercontent.com/assets/15876383/11191019/fd150496-8c66-11e5-9931-a6ac90565a3b.jpg)

**Comment by ntruesdell** [comment]
Several users in our organization (myself included) are experiencing this problem. The toolbar installs without issue. The toolbar shows up under View -> Toolbars. Clicking the menu option once causes nothing to happen. Clicking the menu option a second time bring up a dialog in which IE asks you if you would like to disable the add-on. All of the mentioned group policy settings were set to "Not Configured." I have enabled them as pictured above but this does not appear to have resolved the problem.

**Comment by matatk** [comment]
Thank you to everyone who has provided information on this issue and helped us out by testing things; it is appreciated, as we don't have the same domain set-up so we are unable to reproduce it.  We received some further information from @ntruesdell, which I'll paste in here...

> In an effort to find a temporary workaround for the affected users I tried installing prior versions of the toolbar. 2.0 works perfectly. Neither 2011 or 2012 would install stating that Accessibility_toolbar.dll could not be put in the folder on account of permissions issues. (I tried “run as administrator” when executing the two installers so I really don’t know how I could be getting a permissions issue.)

**Comment by matatk** [comment]
@ntruesdell if you run the WAT installer from the command line, you can use the `/LOG` or `/LOG="filename"` switches to ensure the setup process is logged.  By default the log file will be created in your temporary directory, but for more information please consult [the Inno Setup documentation on command-line switches](http://www.jrsoftware.org/ishelp/index.php?topic=setupcmdline).  This might explain what the "permissions error" is with the DLL in recent versions of WAT.

**Comment by ntruesdell** [comment]
Log file attached. This is for the current version.
[WAT.txt](https://github.com/ThePacielloGroup/WebAccessibilityToolbar/files/121616/WAT.txt)

**Comment by menovak** [comment]
hi Nick,

Thanks for the LOG file.  I did not notice any 'failures' in the log file you shared, but this was for installing the 32 bit version of WAT (e.g., Original Setup EXE: C:\Users\ntruesdell\Desktop\WAT_2015-05-20_32bit.exe).  

Are you on 32 bit versions of Windows 7?

Thanks

Mark

**Comment by menovak** [comment]
hi Nick,

Thanks for the LOG file. I did not notice any 'failures' in the log file
you shared, but this was for installing the 32 bit version of WAT (e.g.,
Original Setup EXE: C:\Users\ntruesdell\Desktop\WAT_2015-05-20_32bit.exe).

Are you on 32 bit versions of Windows 7?

Thanks

Mark

**Comment by ntruesdell** [comment]
I'm using 64-bit Windows. During initial troubleshooting I tried both the 32-bit and 64-bit flavors of the toolbar with the same result. I'm attaching the log file from running the 64-bit installer.
[WAT.txt](https://github.com/ThePacielloGroup/WebAccessibilityToolbar/files/122137/WAT.txt)

**Comment by menovak** [comment]
Hi Nick,

Just a note, for 64 bit Windows, you should choose the 32/64 bit version of WAT.  

I took your install 64 bit log, and using WinMerge compared to an install log I created on 
Windows 8.1 (64 bit), and apart from a few expected differences, time stamps, temp
file locations, etc., the files are identical.   

The only real difference is:

Windows 7 is windows version 6.1.7601 SP1
Windows 8.1 is windows version 6.3.9600

Did you say you had a file installation error message when installing, or am I confusing things.

Mark

**Comment by ntruesdell** [comment]
I had a permissions error when attempting to run the 2011 installer. The current installer has never generated any errors.

**Comment by menovak** [comment]
Hi Nick,

Just a note, for 64 bit Windows, you should choose the 32/64 bit version of
WAT.

I took your install 64 bit log, and using WinMerge compared to an install
log I created on
Windows 8.1 (64 bit), and apart from a few expected differences, time
stamps, temp
file locations, etc., the files are identical.

The only real difference is:

Windows 7 is windows version 6.1.7601 SP1
Windows 8.1 is windows version 6.3.9600

Did you say you had a file installation error message when installing, or
am I confusing things.

Mark

**Comment by menovak** [comment]
Hi Nick,

Sorry for the double emails, thanks for your reply, that helps. 

Mark

**Comment by menovak** [comment]
These are the steps I ran. I was then able to enable the toolbar. Hope this helps
 
 
1)            Close IE
2)            Uninstall add-in if already installed.
3)            Open regedit and navigate to:
HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft\Internet Explorer\Restrictions
4)            Set the NoExtensionManagement key to 0.
5)            Install add-in and Enable it when prompted by IE.
6)            Close IE.
7)            Set NoExtensionManagement key to 1.
8)            Relaunch IE and test, making sure Web Accessibility Toolbar is Checked under View -> Toolbars.

**Comment by SlickVisuals** [comment]
I followed your steps, but don't see "NoExtensionManagement" in Restrictions. I'm having the same issues using the toolbar in IE11 Windows 7 Enterprise 64-bit.
![toolbar-not-working-screenshot](https://cloud.githubusercontent.com/assets/4148812/14617178/405bc202-057b-11e6-81f5-4c8b6432ea01.png)
![toolbar-not-working-screenshot 2](https://cloud.githubusercontent.com/assets/4148812/14617262/8eca53f4-057b-11e6-9c57-5947dba74eb3.png)

**Comment by SlickVisuals** [comment]
Found another fix using the steps above shared by menovak. On step 3, I navigated to the following instead: HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft\Internet Explorer\Main
On step 4, I set "Enable Browser Extensions" from "no" to "yes". THEN installed the add-in. Worked great.

I'm also using Windows 7 Enterprise / 64-bit / 2.0.0.0 version
![toolbar-ie-fix-screenshot](https://cloud.githubusercontent.com/assets/4148812/14617570/b202abb8-057c-11e6-908a-8a5574aed5ae.png)

**Comment by menovak** [comment]
From another trouble shooting adventure... found this:

I found out why the WAT doesn’t work on Windows 10 in our environment.  For the toolbar to work, I found that I had to disable protected mode in IE, change the HKCU Enable Browser Extensions reg key to Yes, and give the user account local admin rights to enable the toolbar.  The problem is that we have Group Policies in place that enables protected mode and doesn’t allow for enabling browser extensions.  So, the issue I was having was caused by restrictions in our Group Policies.
